dbms/src/AggregateFunctions/AggregateFunctionLeastSqr.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <Columns/IColumn.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+template <typename X, typename Y, typename Ret>
+struct AggregateFunctionLeastSqrData
```

Comment:
Maybe, it will be useful to implement non-aggregating format of this functionality with variadic arguments?
```
select leastSqr((1,100), (2,110), (3,120))
```
This may be useful to make regression over groupArray results, for example.